src/systemd-rdma-settings.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+SERV=`echo $1 | awk -F@ '{print $1}'`
+FILE="/usr/lib/systemd/system/"$SERV"@.service"
+ARG_DIR="/run/systemd/system/"$1""
+NAME="$1"
+FLAG=false
+
+add_config(){
+	if [ -f "/run/systemd/system/ceph-osd.target.wants/"$NAME"" ]; then
+		sudo rm -rf "/run/systemd/system/ceph-osd.target.wants/"$NAME""
+	fi
+
+	for CONF in "$@"
+	do
+		if [ ! -f $CONF ]; then
+			touch $CONF
+			FLAG=true
+			cat $FILE > $CONF
+		fi
+		if ! grep -q "[Service]" $CONF ; then
+			echo "[Service]"| sudo tee -a $CONF
+			FLAG=true
+		fi
+		if grep -q "PrivateDevices=yes" $CONF; then
+			sudo sed -i 's/PrivateDevices=yes/PrivateDevices=no/g' $CONF
```

Comment:
It is not the ceph.conf that is being edited. I named the variable CONF since it is a systemd conf file that adds configuration to the original service file. For each service called we create a different conf file in /run/systemd/system/ so it is not possible that two service running in parallel will access the same file.

I can mind changing the name of the variable if you think it is confusing.